actioncable/app/assets/javascripts/action_cable/connection.coffee

```diff
@@ -92,8 +95,15 @@ class ActionCable.Connection
       @disconnected = false
       @subscriptions.reload()
 
-    close: ->
+    close: (event) ->
       ActionCable.log("WebSocket onclose event")
+      if event.code == 1006
+        ActionCable.log("WebSocket received abnormal close error (" + event.code + "): " + event.reason)
+      
+      if !event.wasClean
+        ActionCable.log("WebSocket says close was unclean. Stopping ConnectionMonitor.")
+        @monitor.stop()
+
```

Comment:
Yeah we need a way of telling the app why there was a hard disconnect. To be discussed, as there are other cases where we need this as well (e.g. authentication failure).